src/mlagility/helpers/labels.py

```diff
@@ -0,0 +1,80 @@
+import os
+from typing import Dict, List
+
+
+def load_from_file(file_path: str) -> Dict[str, List[str]]:
+    """
+    This function extracts labels from a Python file.
+    Labels must be in the first line of a Python file and start with "# labels: "
+    Each label must have the format "key::value1,value2,..."
+
+    Example:
+        "# labels: author:google test_group:daily,monthly"
+    """
+    # Labels must be in the first line of the python file
+    with open(file_path, encoding="utf-8") as f:
+        first_line = f.readline()
+
+    if "# labels:" in first_line:
+
+        # Convert labels to dict to ensure they are not malformed
+        label_list = first_line.replace("\n", "").split(" ")[2:]
+        label_dict = convert_to_dict(label_list)
+        return label_dict
+    else:
+        return {}
+
+
+def load_from_cache(cache_dir: str, build_name: str) -> Dict[str, List[str]]:
+    """
+    Loads labels from the cache directory
+    """
+
+    labels_dir = os.path.join(cache_dir, "labels")
+    file_path = os.path.join(labels_dir, f"{build_name}.txt")
+    with open(file_path, encoding="utf-8") as f:
+        first_line = f.readline()
+
+    # Convert labels to dict to ensure they are not malformed
+    label_list = first_line.replace("\n", "").split(" ")
+    label_dict = convert_to_dict(label_list)
+    return label_dict
+
+
+def convert_to_dict(label_list: List[str]) -> Dict[str, List[str]]:
+    """
+    Convert label list into a dictionary of labels
+    """
+    label_dict = {}
+    for item in label_list:
+        try:
+            label_key, label_value = item.split("::")
+        except ValueError:
+            # FIXME: Create a proper warning for this once we have the right
+            # infrastructure for doing so.
+            print(
+                f"Warning: Malformed label {item} found. ",
+                "Each label must have the format key::value1,value2,... ",
+            )
+        label_value = label_value.split(",")
+        label_dict[label_key] = label_value
+    return label_dict
+
+
+def save_to_cache(cache_dir: str, build_name: str, label_dict: Dict[str, List[str]]):
```

Comment:
Since its a dictionary could you save a lot of code by saving the labels as a yaml file instead of a text file?

Reply:
Added an issue for that. Let's try to avoid major breaking changes for now before we integrate this infrastructure with our trackers.

https://github.com/groq/mlagility/issues/56